### PR TITLE
chore(aggregations, find-in-page, import-export, databases-collections): refactor Global.Modal to use new plugin interface COMPASS-7398

### PR DESCRIPTION
**Pull request by gribnoysup**
This patch replaces `Global.Modal` role with plugins registered with the new hadron plugin interface and directly imported in compass-home

The only additional refactoring I did is removing "duplicate view" plugin because it was literally the same plugin as "create view" just registered under a different name, doesn't seem like there is a need for that

### CODE REVIEW

File: packages/databases-collections/package.json [hunk 1]
```diff
@@ -17,14 +17,14 @@
     "dist"
   ],
   "main": "dist/index.js",
-  "compass:main": "src/index.js",
+  "compass:main": "src/index.ts",
```
Comment (addaleax) [inline packages/databases-collections/package.json:20]: :rocket:

**Review by addaleax — APPROVED** [review]

File: packages/compass-aggregations/src/modules/create-view/index.js [hunk 2]
```diff
@@ -158,16 +156,9 @@ export const createView = () => {
       await ds.createView(viewName, viewSource, viewPipeline, options);
       debug('View created!');
       track('Aggregation Saved As View', { num_stages: viewPipeline.length });
-      dispatch(
-        globalAppRegistryEmit('compass:aggregations:create-view', {
-          numStages: viewPipeline.length,
-        })
-      );
```
[the remark this answers is not in the document]
Comment (gribnoysup) [reply packages/compass-aggregations/src/modules/create-view/index.js:165]: Whoops, missed that, thanks!

[the remark this answers is not in the document]
Comment (gribnoysup) [reply packages/compass-aggregations/src/modules/create-view/index.js:165]: Removed in https://github.com/mongodb-js/compass/pull/5081/commits/9a2f59002f24fe787dbff08b922296ada3ac8827

**Review by Anemy — APPROVED** [review]
Nice! I like the removal of extra layers we had for no real reason we had before. And we're now cleaning up our event listeners on deactive ! 🎉

**Pull request merged**